app.js

```diff
@@ -3,87 +3,164 @@ var cheerio = require('cheerio');
-                      authorLink.push($(this).attr('href'));
-                  });
-                  for(i = 0; i < title.length; i++) {
-                      var JSONObjectReview = {reviewer:'',text:'',title:''};
-                      JSONObjectReview.reviewer = author[i];
-                      JSONObjectReview.text = review[i];
-                      JSONObjectReview.title = title[i];
-                      arrayOfReviews.push(JSONObjectReview);
-                  }
-                  completedRequests++;
-                  totalReviews += title.length;
-                  if (completedRequests == totalPages) {
-                      console.log('Total reviews parsed: ' + totalReviews);
-                      resolve(JSON.stringify(arrayOfReviews));
-                  }
-              }
-            })
+  return new Promise(function (resolve,reject) {
+    var totalReviews = 0;
+    var page;
+    var completedRequests = 0;
+    for(page = 1; page < totalPages+1; page++) {
+      cloudscraper.get(review_url + productId + review_page + page, function(error, response, body) {
+        if (error) {
+          console.log('Error occurred');
+          reject(error);
+        } else {
+          var title = [];
+          var review = [];
+          var author = [];
+          var authorLink = [];
+          var $ = cheerio.load(body);
+          $("a[class='a-size-base a-link-normal review-title a-color-base a-text-bold']").each(function(i, element){
+              // console.log((i) + '==>' + $(this).text());
+              title.push($(this).text());
+          });
+          $("span[class='a-size-base review-text']").each(function(i, element){
+              // console.log((i) + '==>' + $(this).text());
+              review.push($(this).text());
+          });
+          $("a[class='a-size-base a-link-normal author']").each(function(i, element){
+              // console.log((i) + '==>' + $(this).text() + ' ' + $(this).attr('href'));
+              author.push($(this).text());
+              authorLink.push($(this).attr('href'));
+          });
+          for(i = 0; i < title.length; i++) {
+            var JSONObjectReview = {reviewer:'',text:'',title:''};
+            JSONObjectReview.reviewer = author[i];
+            JSONObjectReview.text = review[i];
+            JSONObjectReview.title = title[i];
+            arrayOfReviews.push(JSONObjectReview);
+          }
+
+          completedRequests++;
+          totalReviews += title.length;
+          if (completedRequests == totalPages) {
+            console.log('Total reviews parsed: ' + totalReviews);
+            watsonAddDocument(arrayOfReviews).then(discoveryQuery);
+            resolve(JSON.stringify(arrayOfReviews));
+          }
+        }
+      });
+    }
+  });
+}
+
+function watsonAddDocument(reviews) {
+  var promise = new Promise(function (resolve,reject) {
+    var docs = 0;
+    var errorFlag = false;
+    for (var i = 0; i < 2; i++) {
```

Comment:
why less than 2?

Reply:
this is simply for debugging purposes (it's easy to check for uploading two reviews instead of hundreds). This can be changed to all of the parsed reviews in the future.